libfwupdplugin/fu-quirks.c

```diff
@@ -124,70 +124,92 @@ fu_quirks_convert_keyfile_to_xml(FuQuirks *self, GBytes *bytes, GError **error)
 {
 	gsize xmlsz;
 	g_autofree gchar *xml = NULL;
-	g_auto(GStrv) groups = NULL;
-	g_autoptr(GKeyFile) kf = g_key_file_new();
+	g_auto(GStrv) lines = NULL;
+	g_autoptr(GString) group = g_string_new(NULL);
+	g_autoptr(XbBuilderNode) bn = NULL;
 	g_autoptr(XbBuilderNode) root = xb_builder_node_new("quirk");
 
-	/* parse keyfile */
-	if (!g_key_file_load_from_data(kf,
-				       g_bytes_get_data(bytes, NULL),
-				       g_bytes_get_size(bytes),
-				       G_KEY_FILE_NONE,
-				       error))
-		return NULL;
-
-	/* add each set of groups and keys */
-	groups = g_key_file_get_groups(kf, NULL);
-	for (guint i = 0; groups[i] != NULL; i++) {
-		g_auto(GStrv) keys = NULL;
-		g_autofree gchar *group_id = NULL;
+	/* split into lines */
+	lines = fu_strsplit((const gchar *)g_bytes_get_data(bytes, NULL),
```

Comment:
actually, we can do better still; fu_strsplit_full() reduces RSS peak much more.

Reply:
Fixed.